impl-tinkerpop/src/test/java/org/hawkular/inventory/impl/tinkerpop/test/BasicTest.java

```diff
@@ -318,16 +334,149 @@ public void testRelationshipServiceNamed2() throws Exception {
+        // delete the relationship
+        inventory.tenants().get("com.example.tenant").relationships(/*defaults to outgoing*/).delete(link.getId());
+        assert inventory.tenants().get("com.example.tenant").relationships()
+                .named("crossTenantLink").entities().size() == 0 : "Relation 'crossTenantLink' was found.";
+
+        // try deleting again
+        try {
+            inventory.tenants().get("com.example.tenant").relationships(/*defaults to outgoing*/).delete(link.getId());
+            assert false | true : "It shouldn't be possible to delete the same relationship twice";
+        } catch (RelationNotFoundException e) {
+            // good
+        }
+    }
+
+    @Test
+    public void testRelationshipServiceUpdateRelationship() throws Exception {
+        final String someKey = "k3y";
+        final String someValue = "v4lu3";
+        Relationship rel1 = inventory.tenants().get("com.example.tenant").environments().get("test").resources()
+                .get("playroom2").metrics().get("playroom2_size").relationships(Relationships.Direction.outgoing)
+                .named("yourMom").entities().iterator().next();
+        assert null == rel1.getProperties().get(someKey) : "There should not be any property with key 'k3y'";
+        rel1.getProperties().put(someKey, someValue);
+
+        Relationship rel2 = inventory.tenants().get("com.example.tenant").environments().get("test").resources()
+                .get("playroom2").metrics().get("playroom2_size").relationships(Relationships.Direction.outgoing)
+                .named("yourMom").entities().iterator().next();
+        assert rel1.getId().equals(rel2.getId()) && null == rel2.getProperties().get(someKey) : "There should not be" +
+                " any property with key 'k3y'";
+
+        // persist the change
+        inventory.tenants().get("com.example.tenant").environments().get("test").resources()
+                .get("playroom2").metrics().get("playroom2_size").relationships(Relationships.Direction.outgoing)
+                .update(rel1);
+
+        Relationship rel3 = inventory.tenants().get("com.example.tenant").environments().get("test").resources()
+                .get("playroom2").metrics().get("playroom2_size").relationships(Relationships.Direction.outgoing)
+                .named("yourMom").entities().iterator().next();
+        assert rel1.getId().equals(rel3.getId()) && someValue.equals(rel3.getProperties().get(someKey))
+                : "There should be the property with key 'k3y' and value 'v4lu3'";
 
+        try {
+            inventory.tenants().get("com.example.tenant").environments().get("test").resources()
+                    .get("playroom2").metrics().get("playroom1_size").relationships(Relationships.Direction.both)
+                    .update(rel1);
+            assert !!!true : "It shouldn't be possible to update an edge that is not on the current position in the " +
```

Comment:
The test is passing because it throws the `RelationNotFoundException` exception, but from a different reason than the source or target mismatch. It fails because of the wrong rel.id. To be more precise, because of non-existent relation on the current position with such an id.

I'll add another test testing the scenario with the target/source vertex mismatch. You're right that it wasn't working because of calling the `source()` twice, because the `ifThenElse` pipe had no effect, when starting the pipe after few lines from scratch. Good catch!